lib/RWMutex.ts

```diff
@@ -39,14 +39,25 @@ export default class RWMutex {
 
   /*
    * Acquires the write lock.
+   * @param {callback} - Optional callback
    * @return {Promise} - Promise that resolves when the lock is acquired, rejects if an error occurs
    */
-  async lock() {
-    // provides readable error messages, no need to catch and rethrow
-    const lock = await this._findOrCreateLock();
+  async lock(cb) {
+    let lock;
+    try {
+      lock = await this._findOrCreateLock();
+    } catch (err) {
+      if (typeof cb === "function") {
```

Comment:
This is pretty verbose and makes things a bit harder to read and understand. I don't really know best practices here but some questions/suggestions

1) If we don't do this, are there libraries that the consumer can use to simulate this supporting callbacks?
2) Are the libraries we can use so as to not have to do this ourselves (a quick google gave me https://www.npmjs.com/package/callbackify)

If neither of the above are solutions
3) Could we do it a bit less verbosely `typeof cb === "function" && cb(err)`